Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -86,11 +86,14 @@ void HiresTexture::Update()
 		s_textureCache.clear();
 	}
 
-	const std::string& gameCode = SConfig::GetInstance().m_strUniqueID;
+	const std::string& game_id = SConfig::GetInstance().m_strUniqueID;
+	std::string texture_directory = File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id;
```

Comment:
The directory resolution should probably be moved to its own function that returns a `std::string`. Something like GetTextureDirectory(), which would be called like GetTextureDirectory(game_id). Since it isolates it away from the rest of the other logic.